### PR TITLE
Add operator kind to OPERATOR level trace output

### PR DESCRIPTION
**Pull request by virtan**
Summary: Add operator kind to OPERATOR level trace output

Differential Revision: D17643033

### CODE REVIEW

File: tests/unittests/TraceEventsTest.cpp [hunk 1]
```diff
@@ -108,6 +108,36 @@ class TraceEventsTest : public ::testing::TestWithParam<std::string> {
...
+  // In the function below we're trying to compare auto instrumented trace
+  // output with optimized model graph. Since the code generation may make
+  // further changes to the graph representation we limit checking to a set of
+  // convolution / maxpool / cpumaxsplat kinds which are never changed.
+  std::set<std::string> checkedKinds = {"convoluition", "maxpool",
+                                        "cpumaxsplat"};
+
+  void checkEventMetadata(
+      const std::vector<TraceEvent> &traceEvents,
+      std::vector<std::pair<std::string, std::string>> expected) {
```
Comment (rdzhabarov) [inline tests/unittests/TraceEventsTest.cpp]: it's test, does not matter to much, but can you make it `const vector&`

File: tests/unittests/TraceEventsTest.cpp [hunk 2]
```diff
@@ -314,6 +360,21 @@ TEST_P(TraceEventsTest, manualAndAutomatic) {
   cctx.backendOpts.autoInstrument = true;
   EE_.compile(cctx);
 
+  std::vector<std::pair<std::string, std::string>> expectedKinds;
```
Comment (rdzhabarov) [inline tests/unittests/TraceEventsTest.cpp]: can you make this as a helper method which extracts node kinds from the given function and reuse in both tests

File: tests/unittests/TraceEventsTest.cpp (hunk 1 above)
Comment (virtan) [reply tests/unittests/TraceEventsTest.cpp]: I actually repeated the interface of the same function above. But yeah, of course, no any reason to copy this vector here.

File: tests/unittests/TraceEventsTest.cpp (hunk 2 above)
Comment (virtan) [reply tests/unittests/TraceEventsTest.cpp]: Ok

**Pull request closed without merging**